specification/protocol/inference_rest.md

```diff
@@ -0,0 +1,367 @@
+*$metadata_model_error_response* object.
+
+    $metadata_model_error_response =
+    {
+      "error": $string
+    }
+
+* “error” : The descriptive message for the error.
+
+---
+
+### Inference
+
+An inference request is made with an HTTP POST to an inference
+endpoint. In the request the HTTP body contains the
+[Inference Request JSON Object](#inference-request-json-object). In
+the corresponding response the HTTP body contains the
+[Inference Response JSON Object](#inference-response-json-object) or
+[Inference Response JSON Error Object](#inference-response-json-error-object). See
+[Inference Request Examples](#inference-request-examples) for some
+example HTTP/REST requests and responses.
+
+#### Inference Request JSON Object
+
+The inference request object, identified as *$inference_request*, is
+required in the HTTP body of the POST request. The model name and
+(optionally) version must be available in the URL. If a version is not
+provided the server may choose a version based on its own policies or
+return an error.
+
+    $inference_request =
+    {
+      "id" : $string #optional,
+      "parameters" : $parameters #optional,
+      "inputs" : [ $request_input, ... ],
+      "outputs" : [ $request_output, ... ] #optional
+    }
+
+* "id" : An identifier for this request. Optional, but if specified
+  this identifier must be returned in the response.
+* "parameters" : An object containing zero or more parameters for this
+  inference request expressed as key/value pairs. See
+  [Parameters](#parameters) for more information.
+* "inputs" : The input tensors. Each input is described using the
+  *$request_input* schema defined in [Request Input](#inference_request-input).
+* "outputs" : The output tensors requested for this inference. Each
+  requested output is described using the *$request_output* schema
+  defined in [Request Output](#inference_request-output). Optional, if not
+  specified all outputs produced by the model will be returned using
+  default *$request_output* settings.
+
+##### Request Input
+
+The *$inference_request_input* JSON describes an input to the model. If the
+input is batched, the shape and data must represent the full shape and
+contents of the entire batch.
+
+    $inference_request_input =
+    {
+      "name" : $string,
+      "shape" : [ $number, ... ],
+      "datatype"  : $string,
+      "parameters" : $parameters #optional,
+      "data" : $tensor_data
+    }
+
+* "name" : The name of the input tensor.
+* "shape" : The shape of the input tensor. Each dimension must be an
+  integer representable as an unsigned 64-bit integer value.
+* "datatype" : The data-type of the input tensor elements as defined
+  in [Tensor Data Types](#tensor-data-types).
+* "parameters" : An object containing zero or more parameters for this
+  input expressed as key/value pairs. See [Parameters](#parameters)
+  for more information.
+* “data”: The contents of the tensor. See [Tensor Data](#tensor-data)
```

Comment:
Will there be additional parts that explain Tensor Data, Tensor Data Types, Platforms, etc.?

Reply:
Thanks for pointing out, I missed these two sections.